diff --git a/address_dawa.info.yml b/address_dawa.info.yml
index 2b601b6..5f3f1b6 100644
--- a/address_dawa.info.yml
+++ b/address_dawa.info.yml
@@ -7 +7,2 @@ dependencies:
   - drupal:field
+core_version_requirement: ^8 || ^9
diff --git a/tests/src/FunctionalJavascript/AddressDawaFieldTest.php b/tests/src/FunctionalJavascript/AddressDawaFieldTest.php
index 0db9678..1cdbb5d 100644
--- a/tests/src/FunctionalJavascript/AddressDawaFieldTest.php
+++ b/tests/src/FunctionalJavascript/AddressDawaFieldTest.php
@@ -4,3 +4,2 @@ namespace Drupal\Tests\address_dawa\Kernel\Formatter;

-use Drupal\Component\Utility\Unicode;
 use Drupal\entity_test\Entity\EntityTest;
@@ -8,3 +7,3 @@ use Drupal\field\Entity\FieldConfig;
 use Drupal\field\Entity\FieldStorageConfig;
-use Drupal\FunctionalJavascriptTests\JavascriptTestBase;
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
 use Drupal\language\Entity\ConfigurableLanguage;
@@ -16,3 +15,4 @@ use Drupal\language\Entity\ConfigurableLanguage;
  */
-class AddressDawaFieldTest extends JavascriptTestBase {
+class AddressDawaFieldTest extends WebDriverTestBase {
+  protected $defaultTheme = 'stable';

@@ -90,3 +90,3 @@ class AddressDawaFieldTest extends JavascriptTestBase {
     $this->bundle = $this->entityType;
-    $this->fieldName = Unicode::strtolower($this->randomMachineName());
+    $this->fieldName = mb_strtolower($this->randomMachineName());

@@ -105,3 +105,3 @@ class AddressDawaFieldTest extends JavascriptTestBase {
     $this->fieldConfig->save();
-    $this->display = entity_get_display($this->entityType, $this->bundle, 'default');
+    $this->display = \Drupal::service('entity_display.repository')->getViewDisplay($this->entityType, $this->bundle, 'default');
     $this->display->setComponent($this->fieldName, [
@@ -112,3 +112,3 @@ class AddressDawaFieldTest extends JavascriptTestBase {

-    entity_get_form_display($this->fieldConfig->getTargetEntityTypeId(), $this->fieldConfig->getTargetBundle(), 'default')
+    \Drupal::service('entity_display.repository')->getFormDisplay($this->fieldConfig->getTargetEntityTypeId(), $this->fieldConfig->getTargetBundle(), 'default')
       ->setComponent($this->fieldName, array(
