diff --git a/src/Service/ViewsBulkOperationsViewData.php b/src/Service/ViewsBulkOperationsViewData.php
index d72ae49..0c9e274 100644
--- a/src/Service/ViewsBulkOperationsViewData.php
+++ b/src/Service/ViewsBulkOperationsViewData.php
@@ -220,7 +220,7 @@ class ViewsBulkOperationsViewData implements ViewsBulkOperationsViewDataInterfac
       $total_results = $view->total_rows;
     }
 
-    if (!empty($pager_options)) {
+    if (!empty($pager_options) && !$pager_options['items_per_page'] === 0) {
       $this->pagerManager->createPager($pager_options['total_items'], $pager_options['items_per_page'], $pager_options['id']);
     }
 
